src/gcp_scanner/crawl.py

```diff
@@ -368,23 +367,28 @@ def get_bucket_names(project_name: str, credentials: Credentials,
       break
 
     for bucket in response.get("items", []):
-      buckets_dict[bucket["name"]] = (bucket, None)
-      if dump_fd is not None:
-        ret_fields = "nextPageToken,items(name,size,contentType,timeCreated)"
+      buckets_dict[bucket["name"]] = (bucket, [])
 
-        req = service.objects().list(bucket=bucket["name"], fields=ret_fields)
+      # Retrieve root folder information
+      ret_fields = "nextPageToken,items(name,size,contentType,timeCreated)"
 
-        while req:
-          try:
-            resp = req.execute()
-            for item in resp.get("items", []):
+      req = service.objects().list(bucket=bucket["name"], fields=ret_fields)
+
+      while req:
+        try:
+          resp = req.execute()
+          for item in resp.get("items", []):
+            if item["name"] == "":
+              # This is the root folder
+              buckets_dict[bucket["name"]][1].append(item)
+            if dump_fd is not None:
               dump_fd.write(json.dumps(item, indent=2, sort_keys=False))
 
-            req = service.objects().list_next(req, resp)
-          except googleapiclient.errors.HttpError:
-            logging.info("Failed to read the bucket %s", bucket["name"])
-            logging.info(sys.exc_info())
-            break
+          req = service.objects().list_next(req, resp)
+        except googleapiclient.errors.HttpError:
+          logging.info("Failed to read the bucket %s", bucket["name"])
+          logging.info(sys.exc_info())
```

Comment:
For log level info, dumping sys.exc_info seems a bit excessive

Reply:
Yeah, but this information can be useful for debugging purposes, in the future.